src/plugin/context/mod.rs

```diff
@@ -25,13 +27,30 @@ use rapier::geometry::DefaultBroadPhase;
 #[cfg(doc)]
 use crate::prelude::{ImpulseJoint, MultibodyJoint, RevoluteJoint, TypedJoint};
 
+/// Marker component for to access the default [`RapierContext`].
+///
+/// This is used by [`systemparams::DefaultRapierContextAccess`] and other default accesses
+/// to help with getting a reference to the correct RapierContext.
+///
+/// If you're making a library, you might be interested in [`RapierContextEntityLink`]
+/// and leverage a [`Query<&RapierContext>`] to find the correct [`RapierContext`] of an entity.
+#[derive(Component, Reflect, Debug, Clone, Copy)]
+pub struct DefaultRapierContext;
+
+/// This is a component applied to any entity containing a rapier handle component.
+/// The inner Entity referred to has the component [`RapierContext`] responsible for handling
+/// its rapier data.
+#[derive(Component, Reflect, Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct RapierContextEntityLink(pub Entity);
+
 /// The Rapier context, containing all the state of the physics engine.
 #[cfg_attr(feature = "serde-serialize", derive(Serialize, Deserialize))]
-#[derive(Resource)]
+#[derive(Component)]
 pub struct RapierContext {
     /// The island manager, which detects what object is sleeping
     /// (not moving much) to reduce computations.
     pub islands: IslandManager,
+    // FIXME: This should be serialized but a bug prevents it
```

Comment:
Is the bug still present?

Reply:
- It's actually a bug for serde_json, tracked and soon fixed with pull requests linked within https://github.com/dimforge/bevy_rapier/pull/560 ; I'll remove this note from there.